src/sync/outgoingSyncUtils.js

```diff
@@ -73,6 +73,16 @@ const safeGet = (record, path) => {
  */
 const generateSyncData = (settings, recordType, record) => {
   switch (recordType) {
+    case 'IndicatorValue': {
+      return {
+        ID: record.id,
+        facility_ID: settings.get(THIS_STORE_ID),
+        period_ID: record.period.id,
+        column_ID: record.column.id,
+        row_ID: record.row.id,
```

Comment:
is it too OTT to do `record?.period.id` etc?

Reply:
I think we had this convo before (with positions reversed), with the question being whether it is good practice to null-check a non-optional realm field. I don't feel particularly strongly either way:

Pros:
- More "defensive" code.

Cons:
= Implication that `null` is an acceptable assignment, which in this case isn't true.

So, IMO, it's probably better to get an error in this situation rather than silently "fail" and potentially mess up server data?